src/app/views/query-runner/query-input/QueryInput.tsx

```diff
@@ -73,44 +105,140 @@ const QueryInput = (props: IQueryInputProps) => {
     }, 500);
   };
 
+  const handleCopy = () => {
+    copy('share-query-text');
+    trackCopyEvent();
+  };
+
+  const trackCopyEvent = () => {
+    const sanitizedUrl = sanitizeQueryUrl(sampleQuery.sampleUrl);
+    telemetry.trackEvent(eventTypes.BUTTON_CLICK_EVENT,
+      {
+        ComponentName: componentNames.SHARE_QUERY_COPY_BUTTON,
+        QuerySignature: `${sampleQuery.selectedVerb} ${sanitizedUrl}`
+      });
+  }
+
```

Comment:
*suggestion*: For future proofing, let's make the functions/data required by the share query component available to the component itself so that movement from one location to another just requires the movement of the component.